v-next/hardhat/src/internal/builtin-plugins/network-manager/json-rpc-request-modifiers/accounts/automatic-sender-provider.ts

```diff
@@ -8,20 +8,23 @@ import { Sender } from "./sender.js";
  * ensuring dynamic selection of the sender for all JSON-RPC requests without requiring manual input.
  */
 export class AutomaticSender extends Sender {
+  #alreadyFetchedAccounts = false;
```

Comment:
Note that as `#firstAccount` can be undefined, I added a second flag to avoid re-requesting.

Reply:
We could add a test here, I can do it if you want because I exactly know where to add it. It will only take me 5 minutes

Reply:
Sure, but in a follow up PR